lib/settings.js

```diff
@@ -341,7 +341,14 @@ class Settings extends EventEmitter {
 
     this._setSettingsFilePath(filePath);
     const settingsFilePath = this._getSettingsFilePath();
-    jsonfile.writeFileSync(settingsFilePath, this._cacheObj);
+    try {
+      jsonfile.writeFileSync(settingsFilePath, {});
+    } catch (err) {
+      // Cannot read or write file. The user may not have permission to
+      // access the file or directory. Throw error.
+      throw err;
```

Comment:
Wait if we're expecting to throw the exception then there's no need to catch, I thought we wanted to handle this, but if not, we don't need try/catch